cocotb/monitors/avalon.py

```diff
@@ -210,7 +221,10 @@ def valid():
                     self.log.info("Received a packet of %d bytes", len(pkt))
                     self.log.debug(hexdump(str((pkt))))
                     self.channel = channel
-                    self._recv(pkt)
+                    if channel is not None and self.report_channel:
```

Comment:
```suggestion
                    if self.report_channel:
```
`channel` can never be `None` here anyway